process/data_processor.py

```diff
@@ -0,0 +1,155 @@
+"""
+loads and preprocesses the structured log data for anomoly prediction
+"""
+import numpy as np
+import pandas as pd
+import time
+import re
+from collections import OrderedDict
+from collections import Counter
+from scipy.special import expit
+
+
+def collect_event_ids(data_frame):
+    """
+    turns input data_frame into a 2 columned dataframe
+    with columns: BlockId, EventSequence
+    where EventSequence is a list of the events that happened to the block
+    """
+    data_dict = OrderedDict()
+    for _, row in data_frame.iterrows():
+        blk_id_list = re.findall(r"(blk_-?\d+)", row["Content"])
+        blk_id_set = set(blk_id_list)
+        for blk_id in blk_id_set:
+            if not blk_id in data_dict:
+                data_dict[blk_id] = []
+            data_dict[blk_id].append(row["EventId"])
+    data_df = pd.DataFrame(
+        list(data_dict.items()), columns=["BlockId", "EventSequence"]
+    )
+    return data_df
+
+
+class FeatureExtractor(object):
+    """
+    class for fitting and transforming the training set
+    then transforming the testing set
+    """
+
+    def __init__(self):
+        self.mean_vec = None
+        self.idf_vec = None
+        self.events = None
+        self.term_weighting = None
+        self.normalization = None
```

Comment:
if removing normalization, remove this line of code

Reply:
there are also multiple locations where normalization would need to be removed in the code (it is passed into most function args)